sdk/platform/http_client/curl/inc/curl_client.hpp

```diff
@@ -24,16 +25,66 @@ class CurlClient
     {
       return settingUp;
     }
+    if (m_request.getMethod() == azure::core::http::HttpMethod::Put)
+    {
+      curl_easy_setopt(m_p_curl, CURLOPT_PUT, 1L);
+    }
+
+    for (const auto header : m_request.getHeaders())
+    {
+      add_header(header.first, header.second);
+    }
+    if (m_request.getBodyBuffer() != nullptr)
+    {
+      add_header("Content-Length", std::to_string(m_request.getBodyBuffer()->_bodyBufferSize));
+      add_header("Transfer-Encoding", std::string());
+      curl_easy_setopt(m_p_curl, CURLOPT_READFUNCTION, read);
+      curl_easy_setopt(m_p_curl, CURLOPT_READDATA, m_request.getBodyBuffer());
+      curl_easy_setopt(m_p_curl, CURLOPT_TRANSFER_ENCODING, 0); // Disable transfer encoding.
+    }
+    curl_easy_setopt(m_p_curl, CURLOPT_HTTPHEADER, m_slist);
+
+    //curl_easy_setopt(m_p_curl, CURLOPT_PROXY, "127.0.0.1:8888"); curl_easy_setopt(m_p_curl, CURLOPT_SSL_VERIFYPEER, false); // Fiddler
     return curl_easy_perform(m_p_curl);
   }
 
+  void add_header(const std::string& name, const std::string& value)
+  {
+    std::string header(name);
```

Comment:
Danger: You're storing a pointer to destroyed data.

Reply:
`curl_slist_append` copies the string.

Reply:
Hmmm OK.